src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -343,6 +345,32 @@ public void deleteSubscription(String subscriptionId) {
     subscriptionRepository.deleteBySubscriptionId(subscriptionId);
   }
 
+  @Transactional
+  public String terminateSubscription(String subscriptionId, OffsetDateTime terminationDate) {
+    var subscriptionResult = subscriptionRepository.findActiveSubscription(subscriptionId);
+    if (subscriptionResult.isEmpty()) {
+      throw new EntityNotFoundException();
+    }
+
+    var subscription = subscriptionResult.get();
+    subscription.setEndDate(terminationDate);
+
+    OffsetDateTime now = OffsetDateTime.now();
+    // The calculation returns a whole number, representing the number of complete units
+    // between the two temporals. For example, the amount in hours between the times 11:30 and
+    // 12:29 will zero hours as it is one minute short of an hour.
+    var delta = ChronoUnit.HOURS.between(terminationDate, now);
+    if (tagProfile.isProductPAYGEligible(subscription.getSku()) && delta > 0) {
```

Comment:
This `if` statement doesn't work as intended. Reason being that subscription SKU is like `MW01882` where `isProductPAYGEligible` deals with swatch product tags like `rhosak`. So we need to do a couple of things here:
1. Lookup the offering based on the SKU (using `offeringRepository`).
2. Lookup the swatch product tag via offering - the data is in tag profile, but it doesn't look like we made a method for this yet.